packages/marshal/src/dot-membrane.js

```diff
@@ -110,7 +110,7 @@ const makeConverter = (mirrorConverter = undefined) => {
   };
   // We need to pass this while convertYoursToMine is still in temporal
   // dead zone, so we wrap it in convertSlotToVal.
-  const convertSlotToVal = (slot, optIface = undefined) =>
+  const convertSlotToVal = (slot, optIface) =>
```

Comment:
This removal of `= undefined` doesn't even satisfy your stylistic preferences, let alone mine. It is, as you say, "only" documentary in that its removal does not change the observable behavior of the program. However, even your argument against using this documentary signal is that it is redundant with the ts-declaration-in-jsdoc-comment documentary signal, which is absent in this case. Removing this one in the absence of the other one leaves none, or rather, leaves only the remaining weak third signal of the parameter name itself.

```suggestion
  /** @type {import('./types.js').ConvertSlotToVal<any>} */
  const convertSlotToVal = (slot, optIface = undefined) =>
```

I verified that in my vscode, with the addition of this ts-declaration-in-jsdoc-comment, the vscode red squiggle goes away and `optIface` is more accurately typed

![image](https://github.com/endojs/endo/assets/273868/711c970c-9d7a-4f2c-a9e0-e5bde2965e54)

This example is also an example of the other point in that thread. The documentary signal in the ts-declaration-in-jsdoc-comment is distant -- in the `@callback` declaration in `./types.js`, and so it not obviously visible at the site of the definition of this `convertSlotToVal` in `dot-membrabe.js`. With the `= undefined`, someone looking at the code locally in `dot-membrane.js` sees that this parameter is supposed to be optional. With *only* the `/** @type {import('./types.js').ConvertSlotToVal<any>} */` as a local documentary signal, they do not.

Reply:
"that thread" being the internal thread that provoked https://github.com/Agoric/agoric-sdk/issues/8035 . I think this example serves as an adequate example of this point from that thread.

Reply:
Btw, since you're not turning on `noImplicitAny` (thank you!), I don't understand what rule change caused the code here to become a type error. Could you explain? I'd love to have a better predictive model.